include/picongpu/particles/traits/GetExchangeMemCfg.hpp

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2017 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include <pmacc/traits/GetFlagType.hpp>
+#include <pmacc/traits/Resolve.hpp>
+#include <pmacc/traits/HasFlag.hpp>
+
+#include <boost/mpl/if.hpp>
+
+
+namespace picongpu
+{
+namespace traits
+{
+namespace detail
+{
+
+    /** bytes reserved for each communication direction
+     *
+     * This is the default configuration, the buffer sizes are defined in memory.param.
+     */
+    struct DefaultExchangeMemCfg
+    {
+        static constexpr uint32_t BYTES_EXCHANGE_X = picongpu::BYTES_EXCHANGE_X;
+        static constexpr uint32_t BYTES_EXCHANGE_Y = picongpu::BYTES_EXCHANGE_Y;
+        static constexpr uint32_t BYTES_EXCHANGE_Z = picongpu::BYTES_EXCHANGE_Z;
+        static constexpr uint32_t BYTES_CORNER = picongpu::BYTES_CORNER;
+        static constexpr uint32_t BYTES_EDGES = picongpu::BYTES_EDGES;
+    };
+
+} // namespace detail
+
+    /** get a memory configuration for species exchange buffer
+     *
```

Comment:
documentation missing: `If exchangeMemCfg is not defined than the default memory exchange size in memory.param are used`